conn.go

```diff
@@ -666,6 +668,20 @@ func (cn *conn) simpleQuery(q string) (res *rows, err error) {
 	}
 }
 
+type noRows struct{}
+
+var emptyRows noRows
+
+var _ driver.Result = noRows{}
+
+func (noRows) LastInsertId() (int64, error) {
+	return 0, errors.New("no LastInsertId available after empty statement")
+}
+
+func (noRows) RowsAffected() (int64, error) {
+	return 0, errors.New("no RowsAffected available after empty statement")
```

Comment:
I feel like this should either not return an error, or the error should be exported.  Otherwise it's difficult to deal with these if you don't fully control the queries you send.